clients/src/main/java/org/apache/kafka/common/record/RecordBatchIterator.java

```diff
@@ -36,6 +38,8 @@ protected T makeNext() {
             if (batch == null)
                 return allDone();
             return batch;
+        } catch (EOFException e) {
```

Comment:
`LogCleanerTest.testCorruptMessageSizeLargerThanBytesAvailable` was failing after the change to `readNextOffset` in `LogCleaner` due to an unexpected EOF. It seems reasonable to catch it here and raise as a `CorruptRecordException` (which is also a `KafkaException`). This made the test pass.